File: number_validation.php
# Code by Arundhathi B
# php program for validating a number.

<?php

function validateNumber($minNum, $maxNum, $disabledNumbers, $userInput) {


        if ($userInput < $minNum || $userInput > $maxNum) {
            echo "Invalid input.\n";
           return;
        }

        if (in_array($userInput, $disabledNumbers)) {
            echo "$userInput is disabled number.\n";
             $userInput++;
            
          return (validateNumber($minNum, $maxNum, $disabledNumbers, $userInput));

      
        }
        return $userInput;

}

function main() {
    $minNum = intval(readline("Enter the minimum number: "));
    $maxNum = intval(readline("Enter the maximum number: "));
    $disabledNumbersInput = readline("Enter the disabled numbers separated by space: ");
    $disabledNumbers = array_map('intval', explode(' ', $disabledNumbersInput));
    $userInput = readline("Enter a number to validate: ");

    $validatedNumber = validateNumber($minNum, $maxNum, $disabledNumbers, $userInput);
    if(!is_numeric($validatedNumber)){
        echo $validatedNumber;
        return;
    }
    echo "Validated Number: $validatedNumber";
}

main();

?>

File: number_validation.py
# Arundhathi B
# Python program for validating a number.

def validate_number(min_num, max_num, disabled_numbers, num):
        
            if num < min_num or num > max_num:
                print("Invalid input.")
                return
            
            if num not in disabled_numbers:
                print("Validated number:", num)
            
            else:
                print(num, "is disabled number.")
                num += 1
                return validate_number(min_num, max_num, disabled_numbers, num)

def main():
    min_num = int(input("Enter the minimum number: "))
    max_num = int(input("Enter the maximum number: "))
    
    disabled_numbers = set(map(int, input("Enter the disabled numbers separated by space: ").split()))
    num = int(input("Enter a number to validate: "))

    validate_number(min_num, max_num, disabled_numbers, num)


if __name__ == "__main__":
    main()
